### PR TITLE
Voice-call Flow 

### PR DESCRIPTION
**Pull request by SoheilDiyanat**

### CODE REVIEW

File: Examples/VoiceCallFlow/CreateVoiceCallFlow.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using MessageBird.Net.ProxyConfigurationInjector;
+using MessageBird.Objects;
+using MessageBird.Objects.VoiceCalls;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Examples.VoiceCallFlow
+{
+    class CreateVoiceCallFlow
```
Comment (jen-obyrne) [inline Examples/VoiceCallFlow/CreateVoiceCallFlow.cs]: Add the access level here please

File: Examples/VoiceCallFlow/CreateVoiceCallFlow.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        const string YourAccessKey = "YOUR_ACCESS_KEY"; // your access key here.
```
Comment (jen-obyrne) [inline Examples/VoiceCallFlow/CreateVoiceCallFlow.cs]: For consts, best practice is to have all upper case characters - `YOUR_ACCESS_KEY`

File: Examples/VoiceCallFlow/CreateVoiceCallFlow.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        static void Main(string[] args)
```
Comment (jen-obyrne) [inline Examples/VoiceCallFlow/CreateVoiceCallFlow.cs]: Add the access level here too please

File: MessageBird/Client.cs [hunk 4]
```diff
@@ -59,7 +62,7 @@ public Message SendMessage(string originator, string body, long[] msisdns, Messa
             var recipients = new Recipients(msisdns);
             var message = new Message(originator, body, recipients, optionalArguments);
 
-            var messages = new Messages(message);
+            var messages = new Resources.Messages(message);
```
Comment (jen-obyrne) [inline MessageBird/Client.cs]: Can you add a using statement for MessageBird.Resources so you don't have to add the namespace here?

File: MessageBird/Client.cs [hunk 5]
```diff
@@ -372,5 +375,98 @@ public void RemoveContactFromGroup(string groupId, string contactId)
...
+        /// <param name="limit">Set how many records will return from the server</param>
+        /// <param name="offset">Identify the starting point to return rows from a result</param>
+        /// <returns>If successful, this request will return an object with a data, _links and pagination properties.</returns>
+        public VoiceCallFlowList ListVoiceCallFlows(int limit = 20, int offset = 0)
+        {
+            var voiceCallFlowLists = new VoiceCallFlowLists();
+
+            var voiceCallFlowList = (VoiceCallFlowList)voiceCallFlowLists.Object;
+            voiceCallFlowList.Limit = limit;
+            voiceCallFlowList.Offset = offset;
```
Comment (jen-obyrne) [inline MessageBird/Client.cs]: Do you need these lines? Looks like they are not being used

File: MessageBird/Objects/VoiceCalls/VoiceCallFlowList.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
```
Comment (jen-obyrne) [inline MessageBird/Objects/VoiceCalls/VoiceCallFlowList.cs]: Delete the unsused using statements

File: MessageBird/Objects/VoiceCalls/VoiceCallFlowRequest.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (jen-obyrne) [inline MessageBird/Objects/VoiceCalls/VoiceCallFlowRequest.cs]: Remove unused using statements here too please

File: MessageBird/Objects/VoiceCalls/VoiceStep.cs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        [JsonProperty("id")]
+        public Guid Id { get; set; }
+
+        [JsonProperty("action")]
+        public string Action { get; set; }
+
+        [JsonProperty("options")]
+        public VoiceStepOption Options { get; set; }
+    }
+        public class VoiceStepOption
```
Comment (jen-obyrne) [inline MessageBird/Objects/VoiceCalls/VoiceStep.cs]: Fix the spacing here

File: MessageBird/Resources/Conversations/ConversationsResource.cs [hunk 9]
```diff
@@ -3,14 +3,14 @@
 
 namespace MessageBird.Resources.Conversations
 {
-    public class ConversationsResource : Resource
+    public class VoiceCallResource : Resource
```
Comment (jen-obyrne) [inline MessageBird/Resources/Conversations/ConversationsResource.cs]: Mmmm, I don't think this should be renamed? Why are you renaming it? 🤔

File: Tests/UnitTests/MessageBirdUnitTests/Resources/ConversationTest.cs [hunk 10]
```diff
@@ -23,7 +23,7 @@ public void Start()
                 .ThatExpects(
                     @"{""to"":""+31612345678"",""type"":""text"",""content"":{""text"":""Hello!""},""channelId"":""619747f69cf940a98fb443140ce9aed2""}")
                 .AndReturns(filename: "ConversationView.json")
-                .FromEndpoint("POST", "conversations/start", ConversationsResource.ConverstationsBaseUrl)
+                .FromEndpoint("POST", "conversations/start", VoiceCallResource.ConverstationsBaseUrl)
```
Comment (jen-obyrne) [inline Tests/UnitTests/MessageBirdUnitTests/Resources/ConversationTest.cs]: This class name shouldn't be changing, this needs to stay `ConversationsResource`

File: Tests/UnitTests/MessageBirdUnitTests/Resources/VoiceCallFlowTest.cs [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+        {
+            var restClient = MockRestClient
+                .ThatExpects("{\"title\": \"Forward call to 31612345678\",\"record\": true,\"steps\": [ { \"options\": { \"destination\": \"31612345678\" }, \"action\": \"transfer\" }  ] }")
+                .AndReturns("{ \"data\": [ { \"id\": \"voice-call-flow-id\", \"title\": \"Forward call to 31612345678\", \"record\": true, \"steps\": [ { \"id\": \"2fa1383e-6f21-4e6f-8c36-0920c3d0730b\", \"action\": \"transfer\", \"options\": { \"destination\": \"31612345678\" } } ], \"createdAt\": \"2017-03-06T14:52:22Z\", \"updatedAt\": \"2017-03-06T14:52:22Z\" } ], \"_links\": {\"self\": \"/call-flows/de3ed163-d5fc-45f4-b8c4-7eea7458c635\" } }")
+                .FromEndpoint("POST", "call-flows")
+                .Get();
+
+            var client = Client.Create(restClient.Object);
+
+            var newVoiceCallFlow = new VoiceCallFlowRequest();
```
Comment (jen-obyrne) [inline Tests/UnitTests/MessageBirdUnitTests/Resources/VoiceCallFlowTest.cs]: Use object initialization here

File: Tests/UnitTests/MessageBirdUnitTests/Resources/VoiceCallFlowTest.cs [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+        public void Update()
+        {
+            var restClient = MockRestClient
+                .ThatExpects("{ \"title\": \"Updated call flow\", \"steps\": [ { \"options\": { \"destination\": \"31611223344\" }, \"action\": \"transfer\" } ] }")
+                .AndReturns("{ \"data\": [ { \"id\": \"voice-call-flow-id\", \"title\": \"Updated call flow\", \"record\": false, \"steps\": [ { \"id\": \"voice-call-flow-id\", \"action\": \"transfer\", \"options\": { \"destination\": \"31611223344\" } } ], \"createdAt\": \"2017-03-06T13:34:14Z\", \"updatedAt\": \"2017-03-06T15:02:38Z\" } ], \"_links\": { \"self\": \"/call-flows/voice-call-flow-id\" } }")
+                .FromEndpoint("PATCH", "call-flows/voice-call-flow-id")
+                .Get();
+            var client = Client.Create(restClient.Object);
+
+            var voiceCallFlow = new VoiceCallFlowRequest();
```
Comment (jen-obyrne) [inline Tests/UnitTests/MessageBirdUnitTests/Resources/VoiceCallFlowTest.cs]: Object initialization can be used here too

**Comment by SoheilDiyanat** [comment]
All the issues have been resolved.

File: MessageBird/Client.cs [hunk 13]
```diff
@@ -344,7 +346,7 @@ public void AddContactsToGroup(string groupId, IEnumerable<string> contactIds)
 
         /// <summary>
         /// Gets a query string to add contact IDs to a group. This uses a
-        /// specific format: ids[]=foo&ids[]=bar. Given the structure of
+        /// specific format: ids[]=foo & ids[]=bar. Given the structure of
```
Comment (jen-obyrne) [inline MessageBird/Client.cs]: remove the spacing between the params here

File: Tests/UnitTests/MessageBirdUnitTests/Resources/VoiceCallFlowTest.cs [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+                .ThatReturns(filename: "VoiceCallFlowView.json")
+                .FromEndpoint("GET", "call-flows/de3ed163-d5fc-45f4-b8c4-7eea7458c635", VoiceCallFlowsResource.VoiceCallFlowsBaseUrl)
+                .Get();
+            var client = Client.Create(restClient.Object);
+
+            var voiceCallFlow = client.ViewVoiceCallFlow("de3ed163-d5fc-45f4-b8c4-7eea7458c635");
+            restClient.Verify();
+
+            Assert.AreEqual("de3ed163-d5fc-45f4-b8c4-7eea7458c635", voiceCallFlow.Id);
+            //Assert.AreEqual("Forward call to 31611223344", voiceCallFlow.Title);
```
Comment (jen-obyrne) [inline Tests/UnitTests/MessageBirdUnitTests/Resources/VoiceCallFlowTest.cs]: delete commented line

File: MessageBird/Objects/VoiceCalls/VoiceCallFlow.cs [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace MessageBird.Objects.VoiceCalls
+{
+
+    public class Options
+    {
+
+        [JsonProperty("destination")] public string Destination { get; set; }
```
Comment (jen-obyrne) [inline MessageBird/Objects/VoiceCalls/VoiceCallFlow.cs]: Can you reformat this file so the JsonProperty attribute is above the property? 

[JsonProperty("destination")]
public string Destination { get; set; }

**Pull request merged**